web/src/client/network.js

```diff
@@ -23,13 +23,99 @@
 
 import { DBusClient } from "./dbus";
 
+import cockpit from "../lib/cockpit";
+
 const NM_SERVICE_NAME = "org.freedesktop.NetworkManager";
 const NM_IFACE = "org.freedesktop.NetworkManager";
+const NM_SETTINGS_IFACE = "org.freedesktop.NetworkManager.Settings";
+const NM_CONNECTION_IFACE = "org.freedesktop.NetworkManager.Settings.Connection";
+const NM_ACTIVE_CONNECTION_IFACE = "org.freedesktop.NetworkManager.Connection.Active";
+const NM_IP4CONFIG_IFACE = "org.freedesktop.NetworkManager.IP4Config";
+
+const CONNECTION_ADDED = "CONNECTIONS_ADDED";
+const CONNECTION_UPDATED = "CONNECTION_UPDATED";
+const CONNECTION_REMOVED = "CONNECTION_REMOVED";
 
 /**
- * Network client
+ * Enum for the active connection state values
+ *
+ * @readonly
+ * @enum { number }
+ * https://networkmanager.dev/docs/api/latest/nm-dbus-types.html#NMActiveConnectionState
  */
-class NetworkClient {
+const CONNECTION_STATE = {
+  UNKWOWN: 0,
+  ACTIVATING: 1,
+  ACTIVATED: 2,
+  DEACTIVATING: 3,
+  DEACTIVATED: 4
+};
+
+// TODO: document
+const CONNECTION_TYPES = {
+  ETHERNET: "802-3-ethernet",
+  WIFI: "802-11-wireless"
+};
+
+/**
+ * @typedef {object} IPAddress
```

Comment:
Question: should we move types definitions to a separated file?

Reply:
Not needed yet.